[2025-01-29T15:15:09.623+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-01-29T15:15:09.636+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-01-29T15:15:09.643+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-01-29T15:15:09.644+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-01-29T15:15:09.652+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-01-29T15:15:09.657+0000] {standard_task_runner.py:72} INFO - Started process 514 to run task
[2025-01-29T15:15:09.659+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '11', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpkoqi9pn7']
[2025-01-29T15:15:09.662+0000] {standard_task_runner.py:105} INFO - Job 11: Subtask spark_consumer
[2025-01-29T15:15:09.694+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host 21ecc35bc506
[2025-01-29T15:15:09.752+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-01-29T15:15:09.753+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-01-29T15:15:33.869+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/main_pipeline.py", line 115, in spark_consumer
    write_to_postgres(df_parsed)
  File "/opt/airflow/dags/main_pipeline.py", line 140, in write_to_postgres
    hook.insert_rows(
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 657, in insert_rows
    with self._create_autocommit_connection(autocommit) as conn:
  File "/usr/local/lib/python3.11/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 618, in _create_autocommit_connection
    with closing(self.get_conn()) as conn:
                 ^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/postgres/hooks/postgres.py", line 157, in get_conn
    conn = deepcopy(self.connection)
                    ^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 212, in connection
    self._connection = self.get_connection(self.get_conn_id())
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/hooks/base.py", line 83, in get_connection
    conn = Connection.get_connection_from_secrets(conn_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/connection.py", line 537, in get_connection_from_secrets
    raise AirflowNotFoundException(f"The conn_id `{conn_id}` isn't defined")
airflow.exceptions.AirflowNotFoundException: The conn_id `postgres_localohost` isn't defined
[2025-01-29T15:15:33.876+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250129T151509, end_date=20250129T151533
[2025-01-29T15:15:33.893+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-01-29T15:15:33.894+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 11 for task spark_consumer (The conn_id `postgres_localohost` isn't defined; 514)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/main_pipeline.py", line 115, in spark_consumer
    write_to_postgres(df_parsed)
  File "/opt/airflow/dags/main_pipeline.py", line 140, in write_to_postgres
    hook.insert_rows(
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 657, in insert_rows
    with self._create_autocommit_connection(autocommit) as conn:
  File "/usr/local/lib/python3.11/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 618, in _create_autocommit_connection
    with closing(self.get_conn()) as conn:
                 ^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/postgres/hooks/postgres.py", line 157, in get_conn
    conn = deepcopy(self.connection)
                    ^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 212, in connection
    self._connection = self.get_connection(self.get_conn_id())
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/hooks/base.py", line 83, in get_connection
    conn = Connection.get_connection_from_secrets(conn_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/connection.py", line 537, in get_connection_from_secrets
    raise AirflowNotFoundException(f"The conn_id `{conn_id}` isn't defined")
airflow.exceptions.AirflowNotFoundException: The conn_id `postgres_localohost` isn't defined
[2025-01-29T15:15:33.934+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-01-29T15:15:33.953+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-01-29T15:15:33.955+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-01-29T16:01:26.503+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-01-29T16:01:26.517+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-01-29T16:01:26.524+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-01-29T16:01:26.524+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-01-29T16:01:26.534+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-01-29T16:01:26.538+0000] {standard_task_runner.py:72} INFO - Started process 388 to run task
[2025-01-29T16:01:26.541+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '7', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpw1eip4bz']
[2025-01-29T16:01:26.543+0000] {standard_task_runner.py:105} INFO - Job 7: Subtask spark_consumer
[2025-01-29T16:01:26.578+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host 5814ec6b4fbb
[2025-01-29T16:01:26.635+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-01-29T16:01:26.637+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-01-29T16:01:51.525+0000] {base.py:84} INFO - Retrieving connection 'postgres_localhost'
[2025-01-29T16:01:51.532+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/main_pipeline.py", line 115, in spark_consumer
    write_to_postgres(df_parsed)
  File "/opt/airflow/dags/main_pipeline.py", line 140, in write_to_postgres
    hook.insert_rows(
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 657, in insert_rows
    with self._create_autocommit_connection(autocommit) as conn:
  File "/usr/local/lib/python3.11/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 618, in _create_autocommit_connection
    with closing(self.get_conn()) as conn:
                 ^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/postgres/hooks/postgres.py", line 189, in get_conn
    self.conn = psycopg2.connect(**conn_args)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.OperationalError: connection to server at "host.docker.internal" (192.168.65.254), port 5433 failed: FATAL:  password authentication failed for user "***"

[2025-01-29T16:01:51.544+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250129T160126, end_date=20250129T160151
[2025-01-29T16:01:51.553+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-01-29T16:01:51.554+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 7 for task spark_consumer (connection to server at "host.docker.internal" (192.168.65.254), port 5433 failed: FATAL:  password authentication failed for user "***"
; 388)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/main_pipeline.py", line 115, in spark_consumer
    write_to_postgres(df_parsed)
  File "/opt/airflow/dags/main_pipeline.py", line 140, in write_to_postgres
    hook.insert_rows(
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 657, in insert_rows
    with self._create_autocommit_connection(autocommit) as conn:
  File "/usr/local/lib/python3.11/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/common/sql/hooks/sql.py", line 618, in _create_autocommit_connection
    with closing(self.get_conn()) as conn:
                 ^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/postgres/hooks/postgres.py", line 189, in get_conn
    self.conn = psycopg2.connect(**conn_args)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.OperationalError: connection to server at "host.docker.internal" (192.168.65.254), port 5433 failed: FATAL:  password authentication failed for user "***"

[2025-01-29T16:01:51.584+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-01-29T16:01:51.600+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-01-29T16:01:51.603+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T13:22:49.634+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T13:22:49.661+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T13:22:49.673+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T13:22:49.674+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T13:22:49.689+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T13:22:49.696+0000] {standard_task_runner.py:72} INFO - Started process 321 to run task
[2025-02-02T13:22:49.700+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '15', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpyoaox6k7']
[2025-02-02T13:22:49.704+0000] {standard_task_runner.py:105} INFO - Job 15: Subtask spark_consumer
[2025-02-02T13:22:49.770+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host e6ed12b0dfa5
[2025-02-02T13:22:49.851+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T13:22:49.853+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T13:22:55.109+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/main_pipeline.py", line 63, in spark_consumer
    .getOrCreate()
     ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/sql/session.py", line 497, in getOrCreate
    sc = SparkContext.getOrCreate(sparkConf)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/context.py", line 515, in getOrCreate
    SparkContext(conf=conf or SparkConf())
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/context.py", line 201, in __init__
    SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/context.py", line 436, in _ensure_initialized
    SparkContext._gateway = gateway or launch_gateway(conf)
                                       ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/java_gateway.py", line 107, in launch_gateway
    raise PySparkRuntimeError(
pyspark.errors.exceptions.base.PySparkRuntimeError: [JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.
[2025-02-02T13:22:55.128+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T132249, end_date=20250202T132255
[2025-02-02T13:22:55.145+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T13:22:55.147+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 15 for task spark_consumer ([JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.; 321)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/main_pipeline.py", line 63, in spark_consumer
    .getOrCreate()
     ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/sql/session.py", line 497, in getOrCreate
    sc = SparkContext.getOrCreate(sparkConf)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/context.py", line 515, in getOrCreate
    SparkContext(conf=conf or SparkConf())
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/context.py", line 201, in __init__
    SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/context.py", line 436, in _ensure_initialized
    SparkContext._gateway = gateway or launch_gateway(conf)
                                       ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/java_gateway.py", line 107, in launch_gateway
    raise PySparkRuntimeError(
pyspark.errors.exceptions.base.PySparkRuntimeError: [JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.
[2025-02-02T13:22:55.175+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-02-02T13:22:55.193+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T13:22:55.195+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T13:37:08.217+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T13:37:08.229+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T13:37:08.235+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T13:37:08.237+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T13:37:08.245+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T13:37:08.250+0000] {standard_task_runner.py:72} INFO - Started process 464 to run task
[2025-02-02T13:37:08.252+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmp5fvaag8s']
[2025-02-02T13:37:08.254+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask spark_consumer
[2025-02-02T13:37:08.287+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host 64f2a5764477
[2025-02-02T13:37:08.343+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T13:37:08.344+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T13:37:19.549+0000] {base.py:84} INFO - Retrieving connection 'postgres_localhost'
[2025-02-02T13:37:20.330+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T13:37:21.107+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T13:37:21.859+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T13:37:22.593+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T13:37:23.315+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T13:37:23.930+0000] {sql.py:681} INFO - Loaded 840 rows into crypto_data so far
[2025-02-02T13:37:23.931+0000] {sql.py:694} INFO - Done loading. Loaded a total of 5840 rows into crypto_data
[2025-02-02T13:37:23.934+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T13:37:23.941+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T13:37:23.942+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T133708, end_date=20250202T133723
[2025-02-02T13:37:23.983+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T13:37:23.994+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T13:37:23.996+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T13:50:34.182+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T13:50:34.195+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T13:50:34.201+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T13:50:34.202+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T13:50:34.210+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T13:50:34.214+0000] {standard_task_runner.py:72} INFO - Started process 1660 to run task
[2025-02-02T13:50:34.216+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '13', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpf8xckryx']
[2025-02-02T13:50:34.218+0000] {standard_task_runner.py:105} INFO - Job 13: Subtask spark_consumer
[2025-02-02T13:50:34.251+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host 774492c58bf0
[2025-02-02T13:50:34.304+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T13:50:34.306+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T13:50:38.760+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/main_pipeline.py", line 70, in spark_consumer
    .load()
     ^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/sql/readwriter.py", line 314, in load
    return self._df(self._jreader.load())
                    ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/py4j/java_gateway.py", line 1322, in __call__
    return_value = get_return_value(
                   ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/errors/exceptions/captured.py", line 185, in deco
    raise converted from None
pyspark.errors.exceptions.captured.IllegalArgumentException: starting offset can't be latest for batch queries on Kafka
[2025-02-02T13:50:38.778+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T135034, end_date=20250202T135038
[2025-02-02T13:50:38.799+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T13:50:38.801+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 13 for task spark_consumer (starting offset can't be latest for batch queries on Kafka; 1660)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/main_pipeline.py", line 70, in spark_consumer
    .load()
     ^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/sql/readwriter.py", line 314, in load
    return self._df(self._jreader.load())
                    ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/py4j/java_gateway.py", line 1322, in __call__
    return_value = get_return_value(
                   ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/pyspark/errors/exceptions/captured.py", line 185, in deco
    raise converted from None
pyspark.errors.exceptions.captured.IllegalArgumentException: starting offset can't be latest for batch queries on Kafka
[2025-02-02T13:50:38.842+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-02-02T13:50:38.861+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T13:50:38.863+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T13:58:43.212+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T13:58:43.224+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T13:58:43.231+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T13:58:43.232+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T13:58:43.240+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T13:58:43.245+0000] {standard_task_runner.py:72} INFO - Started process 463 to run task
[2025-02-02T13:58:43.247+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpqnemwqzv']
[2025-02-02T13:58:43.250+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask spark_consumer
[2025-02-02T13:58:43.283+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host f0b94dd433bd
[2025-02-02T13:58:43.346+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T13:58:43.348+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T13:58:54.584+0000] {base.py:84} INFO - Retrieving connection 'postgres_localhost'
[2025-02-02T13:58:55.420+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T13:58:56.330+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T13:58:57.240+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T13:58:58.115+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T13:58:58.945+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T13:58:59.578+0000] {sql.py:681} INFO - Loaded 840 rows into crypto_data so far
[2025-02-02T13:58:59.579+0000] {sql.py:694} INFO - Done loading. Loaded a total of 5840 rows into crypto_data
[2025-02-02T13:58:59.582+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T13:58:59.590+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T13:58:59.591+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T135843, end_date=20250202T135859
[2025-02-02T13:58:59.641+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T13:58:59.652+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T13:58:59.654+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T14:17:46.353+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T14:17:46.365+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T14:17:46.372+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T14:17:46.373+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T14:17:46.383+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T14:17:46.387+0000] {standard_task_runner.py:72} INFO - Started process 1660 to run task
[2025-02-02T14:17:46.389+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '12', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmprh6i8fj7']
[2025-02-02T14:17:46.391+0000] {standard_task_runner.py:105} INFO - Job 12: Subtask spark_consumer
[2025-02-02T14:17:46.431+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host 5ab92388ff22
[2025-02-02T14:17:46.491+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T14:17:46.492+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T14:17:57.279+0000] {base.py:84} INFO - Retrieving connection 'postgres_localhost'
[2025-02-02T14:17:58.096+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T14:17:58.972+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T14:17:59.899+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T14:18:00.722+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T14:18:01.499+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T14:18:02.145+0000] {sql.py:681} INFO - Loaded 840 rows into crypto_data so far
[2025-02-02T14:18:02.146+0000] {sql.py:694} INFO - Done loading. Loaded a total of 5840 rows into crypto_data
[2025-02-02T14:18:02.149+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T14:18:02.157+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T14:18:02.158+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T141746, end_date=20250202T141802
[2025-02-02T14:18:02.176+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T14:18:02.186+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T14:18:02.188+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T14:36:27.093+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T14:36:27.104+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T14:36:27.109+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T14:36:27.110+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T14:36:27.118+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T14:36:27.122+0000] {standard_task_runner.py:72} INFO - Started process 602 to run task
[2025-02-02T14:36:27.124+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpuwc11vr2']
[2025-02-02T14:36:27.126+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask spark_consumer
[2025-02-02T14:36:27.158+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host be62f1cf8682
[2025-02-02T14:36:27.217+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T14:36:27.218+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T14:36:32.019+0000] {java_gateway.py:2273} INFO - Callback Server Starting
[2025-02-02T14:36:32.020+0000] {java_gateway.py:2275} INFO - Socket listening on ('127.0.0.1', 40623)
[2025-02-02T14:36:33.801+0000] {clientserver.py:561} INFO - Python Server ready to receive messages
[2025-02-02T14:36:33.802+0000] {clientserver.py:575} INFO - Received command c on object id p0
[2025-02-02T14:36:33.965+0000] {base.py:84} INFO - Retrieving connection 'postgres_localhost'
[2025-02-02T14:36:33.973+0000] {sql.py:694} INFO - Done loading. Loaded a total of 0 rows into crypto_data
[2025-02-02T14:37:32.211+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T14:37:32.220+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T14:37:32.221+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T143627, end_date=20250202T143732
[2025-02-02T14:37:32.263+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T14:37:32.274+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T14:37:32.276+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T14:46:48.163+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T14:46:48.174+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T14:46:48.179+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T14:46:48.180+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T14:46:48.188+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T14:46:48.192+0000] {standard_task_runner.py:72} INFO - Started process 471 to run task
[2025-02-02T14:46:48.194+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpdact4f3z']
[2025-02-02T14:46:48.196+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask spark_consumer
[2025-02-02T14:46:48.226+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host 63894a7c3fe6
[2025-02-02T14:46:48.278+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T14:46:48.280+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T14:46:58.287+0000] {base.py:84} INFO - Retrieving connection 'postgres_localhost'
[2025-02-02T14:46:59.055+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T14:46:59.831+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T14:47:00.597+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T14:47:01.312+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T14:47:02.053+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T14:47:02.682+0000] {sql.py:681} INFO - Loaded 840 rows into crypto_data so far
[2025-02-02T14:47:02.683+0000] {sql.py:694} INFO - Done loading. Loaded a total of 5840 rows into crypto_data
[2025-02-02T14:47:02.685+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T14:47:02.693+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T14:47:02.693+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T144648, end_date=20250202T144702
[2025-02-02T14:47:02.721+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T14:47:02.734+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T14:47:02.736+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T14:59:01.562+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T14:59:01.576+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T14:59:01.583+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T14:59:01.584+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T14:59:01.593+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T14:59:01.599+0000] {standard_task_runner.py:72} INFO - Started process 454 to run task
[2025-02-02T14:59:01.601+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpgzk9uadk']
[2025-02-02T14:59:01.604+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask spark_consumer
[2025-02-02T14:59:01.642+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host a75ab04d6f73
[2025-02-02T14:59:01.757+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T14:59:01.759+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T14:59:01.778+0000] {conn.py:380} INFO - <BrokerConnection node_id=bootstrap-0 host=kafka:9093 <connecting> [IPv4 ('172.18.0.5', 9093)]>: connecting to kafka:9093 [('172.18.0.5', 9093) IPv4]
[2025-02-02T14:59:01.780+0000] {conn.py:1205} INFO - Probing node bootstrap-0 broker version
[2025-02-02T14:59:01.782+0000] {conn.py:410} INFO - <BrokerConnection node_id=bootstrap-0 host=kafka:9093 <connecting> [IPv4 ('172.18.0.5', 9093)]>: Connection complete.
[2025-02-02T14:59:01.886+0000] {conn.py:1267} INFO - Broker version identified as 2.5.0
[2025-02-02T14:59:01.887+0000] {conn.py:1268} INFO - Set configuration api_version=(2, 5, 0) to skip auto check_version requests on startup
[2025-02-02T14:59:01.889+0000] {subscription_state.py:171} INFO - Updating subscribed topics to: ('crypto_topic',)
[2025-02-02T14:59:06.902+0000] {cluster.py:371} INFO - Group coordinator for crypto_consumer_group is ('coordinator-1001', 'kafka', 9093, None)
[2025-02-02T14:59:06.903+0000] {base.py:693} INFO - Discovered coordinator coordinator-1001 for group crypto_consumer_group
[2025-02-02T14:59:06.904+0000] {base.py:741} INFO - Starting new heartbeat thread
[2025-02-02T14:59:06.905+0000] {consumer.py:348} INFO - Revoking previously assigned partitions () for group crypto_consumer_group
[2025-02-02T14:59:06.907+0000] {conn.py:380} INFO - <BrokerConnection node_id=coordinator-1001 host=kafka:9093 <connecting> [IPv4 ('172.18.0.5', 9093)]>: connecting to kafka:9093 [('172.18.0.5', 9093) IPv4]
[2025-02-02T14:59:06.909+0000] {conn.py:410} INFO - <BrokerConnection node_id=coordinator-1001 host=kafka:9093 <connecting> [IPv4 ('172.18.0.5', 9093)]>: Connection complete.
[2025-02-02T14:59:06.910+0000] {conn.py:919} INFO - <BrokerConnection node_id=bootstrap-0 host=kafka:9093 <connected> [IPv4 ('172.18.0.5', 9093)]>: Closing connection. 
[2025-02-02T14:59:07.012+0000] {base.py:450} INFO - (Re-)joining group crypto_consumer_group
[2025-02-02T14:59:07.016+0000] {base.py:521} INFO - Elected group leader -- performing partition assignments using range
[2025-02-02T14:59:07.019+0000] {conn.py:380} INFO - <BrokerConnection node_id=1001 host=kafka:9093 <connecting> [IPv4 ('172.18.0.5', 9093)]>: connecting to kafka:9093 [('172.18.0.5', 9093) IPv4]
[2025-02-02T14:59:07.020+0000] {conn.py:410} INFO - <BrokerConnection node_id=1001 host=kafka:9093 <connecting> [IPv4 ('172.18.0.5', 9093)]>: Connection complete.
[2025-02-02T14:59:07.024+0000] {base.py:335} INFO - Successfully joined group crypto_consumer_group with generation 3
[2025-02-02T14:59:07.025+0000] {subscription_state.py:257} INFO - Updated partition assignment: [('crypto_topic', 0)]
[2025-02-02T14:59:07.026+0000] {consumer.py:245} INFO - Setting newly assigned partitions (('crypto_topic', 0),) for group crypto_consumer_group
[2025-02-02T14:59:11.901+0000] {base.py:748} INFO - Stopping heartbeat thread
[2025-02-02T14:59:11.903+0000] {base.py:773} INFO - Leaving consumer group (crypto_consumer_group).
[2025-02-02T14:59:11.906+0000] {conn.py:919} INFO - <BrokerConnection node_id=coordinator-1001 host=kafka:9093 <connected> [IPv4 ('172.18.0.5', 9093)]>: Closing connection. 
[2025-02-02T14:59:11.907+0000] {conn.py:919} INFO - <BrokerConnection node_id=1001 host=kafka:9093 <connected> [IPv4 ('172.18.0.5', 9093)]>: Closing connection. 
[2025-02-02T14:59:11.908+0000] {future.py:79} ERROR - Fetch to node 1001 failed: Cancelled: <BrokerConnection node_id=1001 host=kafka:9093 <connected> [IPv4 ('172.18.0.5', 9093)]>
[2025-02-02T14:59:11.909+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T14:59:11.918+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T14:59:11.919+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T145901, end_date=20250202T145911
[2025-02-02T14:59:11.957+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T14:59:11.967+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T14:59:11.969+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T15:48:12.499+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T15:48:12.511+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T15:48:12.519+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T15:48:12.520+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T15:48:12.530+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T15:48:12.534+0000] {standard_task_runner.py:72} INFO - Started process 2142 to run task
[2025-02-02T15:48:12.536+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '12', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpza355rkw']
[2025-02-02T15:48:12.538+0000] {standard_task_runner.py:105} INFO - Job 12: Subtask spark_consumer
[2025-02-02T15:48:12.575+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host d42d69d5511c
[2025-02-02T15:48:12.636+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T15:48:12.637+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T15:48:17.230+0000] {logging_mixin.py:190} INFO - root
 |-- key: binary (nullable = true)
 |-- value: binary (nullable = true)
 |-- topic: string (nullable = true)
 |-- partition: integer (nullable = true)
 |-- offset: long (nullable = true)
 |-- timestamp: timestamp (nullable = true)
 |-- timestampType: integer (nullable = true)
[2025-02-02T15:48:17.462+0000] {logging_mixin.py:190} INFO - root
 |-- symbol: string (nullable = true)
 |-- priceChange: decimal(20,8) (nullable = true)
 |-- priceChangePercent: decimal(20,8) (nullable = true)
 |-- weightedAvgPrice: decimal(20,8) (nullable = true)
 |-- prevClosePrice: decimal(20,8) (nullable = true)
 |-- lastPrice: decimal(20,8) (nullable = true)
 |-- lastQty: decimal(20,8) (nullable = true)
 |-- bidPrice: decimal(20,8) (nullable = true)
 |-- bidQty: decimal(20,8) (nullable = true)
 |-- askPrice: decimal(20,8) (nullable = true)
 |-- askQty: decimal(20,8) (nullable = true)
 |-- openPrice: decimal(20,8) (nullable = true)
 |-- highPrice: decimal(20,8) (nullable = true)
 |-- lowPrice: decimal(20,8) (nullable = true)
 |-- volume: decimal(20,8) (nullable = true)
 |-- quoteVolume: decimal(20,8) (nullable = true)
 |-- openTime: long (nullable = true)
 |-- closeTime: long (nullable = true)
 |-- firstId: long (nullable = true)
 |-- lastId: long (nullable = true)
 |-- count: long (nullable = true)
[2025-02-02T15:48:17.565+0000] {java_gateway.py:2273} INFO - Callback Server Starting
[2025-02-02T15:48:17.566+0000] {java_gateway.py:2275} INFO - Socket listening on ('127.0.0.1', 38607)
[2025-02-02T15:48:17.794+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T15:48:17.807+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T15:48:17.808+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T154812, end_date=20250202T154817
[2025-02-02T15:48:17.836+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T15:48:17.848+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T15:48:17.850+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T15:54:55.529+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T15:54:55.542+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T15:54:55.549+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T15:54:55.550+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T15:54:55.559+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T15:54:55.563+0000] {standard_task_runner.py:72} INFO - Started process 550 to run task
[2025-02-02T15:54:55.565+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpyxhvhcdl']
[2025-02-02T15:54:55.568+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask spark_consumer
[2025-02-02T15:54:55.599+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host 7bf4a44673c6
[2025-02-02T15:54:55.662+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T15:54:55.663+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T15:55:00.678+0000] {logging_mixin.py:190} INFO - root
 |-- key: binary (nullable = true)
 |-- value: binary (nullable = true)
 |-- topic: string (nullable = true)
 |-- partition: integer (nullable = true)
 |-- offset: long (nullable = true)
 |-- timestamp: timestamp (nullable = true)
 |-- timestampType: integer (nullable = true)
[2025-02-02T15:55:00.910+0000] {logging_mixin.py:190} INFO - root
 |-- symbol: string (nullable = true)
 |-- priceChange: decimal(20,8) (nullable = true)
 |-- priceChangePercent: decimal(20,8) (nullable = true)
 |-- weightedAvgPrice: decimal(20,8) (nullable = true)
 |-- prevClosePrice: decimal(20,8) (nullable = true)
 |-- lastPrice: decimal(20,8) (nullable = true)
 |-- lastQty: decimal(20,8) (nullable = true)
 |-- bidPrice: decimal(20,8) (nullable = true)
 |-- bidQty: decimal(20,8) (nullable = true)
 |-- askPrice: decimal(20,8) (nullable = true)
 |-- askQty: decimal(20,8) (nullable = true)
 |-- openPrice: decimal(20,8) (nullable = true)
 |-- highPrice: decimal(20,8) (nullable = true)
 |-- lowPrice: decimal(20,8) (nullable = true)
 |-- volume: decimal(20,8) (nullable = true)
 |-- quoteVolume: decimal(20,8) (nullable = true)
 |-- openTime: long (nullable = true)
 |-- closeTime: long (nullable = true)
 |-- firstId: long (nullable = true)
 |-- lastId: long (nullable = true)
 |-- count: long (nullable = true)
[2025-02-02T15:55:01.022+0000] {java_gateway.py:2273} INFO - Callback Server Starting
[2025-02-02T15:55:01.023+0000] {java_gateway.py:2275} INFO - Socket listening on ('127.0.0.1', 44719)
[2025-02-02T15:55:01.276+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T15:55:01.289+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T15:55:01.290+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T155455, end_date=20250202T155501
[2025-02-02T15:55:01.346+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T15:55:01.354+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T16:39:14.629+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T16:39:14.642+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T16:39:14.652+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T16:39:14.654+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T16:39:14.671+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T16:39:14.676+0000] {standard_task_runner.py:72} INFO - Started process 2112 to run task
[2025-02-02T16:39:14.678+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '12', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpktro5nvv']
[2025-02-02T16:39:14.681+0000] {standard_task_runner.py:105} INFO - Job 12: Subtask spark_consumer
[2025-02-02T16:39:14.719+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host 4061bda98b6f
[2025-02-02T16:39:14.789+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T16:39:14.790+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T16:39:25.485+0000] {base.py:84} INFO - Retrieving connection 'postgres_localhost'
[2025-02-02T16:39:28.981+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T16:39:30.313+0000] {sql.py:681} INFO - Loaded 1000 rows into crypto_data so far
[2025-02-02T16:39:31.036+0000] {sql.py:681} INFO - Loaded 920 rows into crypto_data so far
[2025-02-02T16:39:31.037+0000] {sql.py:694} INFO - Done loading. Loaded a total of 2920 rows into crypto_data
[2025-02-02T16:39:31.039+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T16:39:31.047+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T16:39:31.048+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T163914, end_date=20250202T163931
[2025-02-02T16:39:31.072+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T16:39:31.086+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T16:39:31.088+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T18:40:26.944+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T18:40:26.956+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T18:40:26.963+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T18:40:26.964+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T18:40:26.972+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T18:40:26.976+0000] {standard_task_runner.py:72} INFO - Started process 1841 to run task
[2025-02-02T18:40:26.978+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '13', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpby5ldfhz']
[2025-02-02T18:40:26.980+0000] {standard_task_runner.py:105} INFO - Job 13: Subtask spark_consumer
[2025-02-02T18:40:27.012+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host 15904aeddad7
[2025-02-02T18:40:27.067+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T18:40:27.068+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T18:40:32.884+0000] {java_gateway.py:2273} INFO - Callback Server Starting
[2025-02-02T18:40:32.885+0000] {java_gateway.py:2275} INFO - Socket listening on ('127.0.0.1', 43253)
[2025-02-02T18:40:35.076+0000] {clientserver.py:561} INFO - Python Server ready to receive messages
[2025-02-02T18:40:35.077+0000] {clientserver.py:575} INFO - Received command c on object id p0
[2025-02-02T18:41:33.123+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T18:41:33.133+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T18:41:33.134+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T184026, end_date=20250202T184133
[2025-02-02T18:41:33.183+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T18:41:33.195+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T18:41:33.196+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T19:20:45.506+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T19:20:45.519+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T19:20:45.526+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T19:20:45.527+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T19:20:45.536+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T19:20:45.541+0000] {standard_task_runner.py:72} INFO - Started process 563 to run task
[2025-02-02T19:20:45.543+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmp3wprhsx7']
[2025-02-02T19:20:45.546+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask spark_consumer
[2025-02-02T19:20:45.582+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host dfbcbb9a3a68
[2025-02-02T19:20:45.641+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T19:20:45.642+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T19:20:50.119+0000] {main_pipeline.py:73} INFO - DataFrame является потоком
[2025-02-02T19:20:50.644+0000] {main_pipeline.py:149} INFO - df_parsed является потоком
[2025-02-02T19:20:50.661+0000] {java_gateway.py:2273} INFO - Callback Server Starting
[2025-02-02T19:20:50.663+0000] {java_gateway.py:2275} INFO - Socket listening on ('127.0.0.1', 45167)
[2025-02-02T19:20:52.573+0000] {clientserver.py:561} INFO - Python Server ready to receive messages
[2025-02-02T19:20:52.574+0000] {clientserver.py:575} INFO - Received command c on object id p0
[2025-02-02T19:20:53.574+0000] {main_pipeline.py:175} INFO - Запись пакета 0 с данными: Row(symbol='ETHBTC', price_change=Decimal('-0.00165000'), price_change_percent=Decimal('-5.22000000'), weighted_avg_price=Decimal('0.03083062'), prev_close_price=Decimal('0.03160000'), last_price=Decimal('0.02996000'), last_qty=Decimal('3.43440000'), bid_price=Decimal('0.02996000'), bid_qty=Decimal('25.82290000'), ask_price=Decimal('0.02997000'), ask_qty=Decimal('21.77020000'), open_price=Decimal('0.03161000'), high_price=Decimal('0.03164000'), low_price=Decimal('0.02983000'), volume=Decimal('37675.78250000'), quote_volume=Decimal('1161.56779859'), close_time=datetime.datetime(2025, 2, 2, 19, 19, 50), first_id=485441387, last_id=485575574, count=134188, trend=False) в PostgreSQL.
[2025-02-02T19:20:53.575+0000] {main_pipeline.py:178} INFO - Writing batch 0 to PostgreSQL.
[2025-02-02T19:20:53.632+0000] {main_pipeline.py:179} INFO - Writing batch with data: Row(symbol='ETHBTC', price_change=Decimal('-0.00165000'), price_change_percent=Decimal('-5.22000000'), weighted_avg_price=Decimal('0.03083062'), prev_close_price=Decimal('0.03160000'), last_price=Decimal('0.02996000'), last_qty=Decimal('3.43440000'), bid_price=Decimal('0.02996000'), bid_qty=Decimal('25.82290000'), ask_price=Decimal('0.02997000'), ask_qty=Decimal('21.77020000'), open_price=Decimal('0.03161000'), high_price=Decimal('0.03164000'), low_price=Decimal('0.02983000'), volume=Decimal('37675.78250000'), quote_volume=Decimal('1161.56779859'), close_time=datetime.datetime(2025, 2, 2, 19, 19, 50), first_id=485441387, last_id=485575574, count=134188, trend=False) to PostgreSQL.
[2025-02-02T19:20:55.903+0000] {main_pipeline.py:181} INFO - Batch 0 successfully written to PostgreSQL.
[2025-02-02T19:21:10.886+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T19:21:10.896+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T19:21:10.898+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T192045, end_date=20250202T192110
[2025-02-02T19:21:10.935+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T19:21:10.951+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T19:21:10.953+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T19:53:56.235+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T19:53:56.247+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T19:53:56.254+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T19:53:56.254+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T19:53:56.264+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T19:53:56.268+0000] {standard_task_runner.py:72} INFO - Started process 569 to run task
[2025-02-02T19:53:56.271+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmp9nh3yfip']
[2025-02-02T19:53:56.273+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask spark_consumer
[2025-02-02T19:53:56.307+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host cc19208c71dd
[2025-02-02T19:53:56.375+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T19:53:56.377+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T19:54:03.684+0000] {main_pipeline.py:73} INFO - DataFrame является потоком
[2025-02-02T19:54:04.545+0000] {main_pipeline.py:149} INFO - df_parsed является потоком
[2025-02-02T19:54:04.558+0000] {java_gateway.py:2273} INFO - Callback Server Starting
[2025-02-02T19:54:04.565+0000] {java_gateway.py:2275} INFO - Socket listening on ('127.0.0.1', 39031)
[2025-02-02T19:54:07.388+0000] {clientserver.py:561} INFO - Python Server ready to receive messages
[2025-02-02T19:54:07.390+0000] {clientserver.py:575} INFO - Received command c on object id p0
[2025-02-02T19:54:08.576+0000] {main_pipeline.py:175} INFO - Запись пакета 0 с данными: Row(symbol='ETHBTC', price_change=Decimal('-0.00123000'), price_change_percent=Decimal('-3.91300000'), weighted_avg_price=Decimal('0.03073475'), prev_close_price=Decimal('0.03144000'), last_price=Decimal('0.03020000'), last_qty=Decimal('2.63620000'), bid_price=Decimal('0.03019000'), bid_qty=Decimal('18.51300000'), ask_price=Decimal('0.03020000'), ask_qty=Decimal('37.50540000'), open_price=Decimal('0.03143000'), high_price=Decimal('0.03146000'), low_price=Decimal('0.02980600'), volume=Decimal('39308.29160000'), quote_volume=Decimal('1208.13059068'), close_time=datetime.datetime(2025, 2, 2, 19, 52, 55), first_id=485448302, last_id=485584595, count=136294, trend=False) в PostgreSQL.
[2025-02-02T19:54:08.578+0000] {main_pipeline.py:178} INFO - Writing batch 0 to PostgreSQL.
[2025-02-02T19:54:08.649+0000] {main_pipeline.py:179} INFO - Writing batch with data: Row(symbol='ETHBTC', price_change=Decimal('-0.00123000'), price_change_percent=Decimal('-3.91300000'), weighted_avg_price=Decimal('0.03073475'), prev_close_price=Decimal('0.03144000'), last_price=Decimal('0.03020000'), last_qty=Decimal('2.63620000'), bid_price=Decimal('0.03019000'), bid_qty=Decimal('18.51300000'), ask_price=Decimal('0.03020000'), ask_qty=Decimal('37.50540000'), open_price=Decimal('0.03143000'), high_price=Decimal('0.03146000'), low_price=Decimal('0.02980600'), volume=Decimal('39308.29160000'), quote_volume=Decimal('1208.13059068'), close_time=datetime.datetime(2025, 2, 2, 19, 52, 55), first_id=485448302, last_id=485584595, count=136294, trend=False) to PostgreSQL.
[2025-02-02T19:54:12.028+0000] {main_pipeline.py:181} INFO - Batch 0 successfully written to PostgreSQL.
[2025-02-02T19:54:24.889+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T19:54:24.901+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T19:54:24.902+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T195356, end_date=20250202T195424
[2025-02-02T19:54:24.958+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T19:54:24.975+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T19:54:24.978+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T20:09:16.122+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T20:09:16.135+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T20:09:16.142+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T20:09:16.143+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T20:09:16.152+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T20:09:16.156+0000] {standard_task_runner.py:72} INFO - Started process 560 to run task
[2025-02-02T20:09:16.159+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpvb3e_uqs']
[2025-02-02T20:09:16.162+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask spark_consumer
[2025-02-02T20:09:16.197+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host 6df1dc328a9b
[2025-02-02T20:09:16.261+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T20:09:16.262+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T20:09:22.153+0000] {main_pipeline.py:143} INFO - df_parsed является потоком
[2025-02-02T20:09:22.167+0000] {java_gateway.py:2273} INFO - Callback Server Starting
[2025-02-02T20:09:22.168+0000] {java_gateway.py:2275} INFO - Socket listening on ('127.0.0.1', 36273)
[2025-02-02T20:09:25.180+0000] {clientserver.py:561} INFO - Python Server ready to receive messages
[2025-02-02T20:09:25.182+0000] {clientserver.py:575} INFO - Received command c on object id p0
[2025-02-02T20:09:25.224+0000] {main_pipeline.py:167} INFO - Batch 0 пустой.
[2025-02-02T20:09:25.225+0000] {main_pipeline.py:172} INFO - Writing batch 0 to PostgreSQL.
[2025-02-02T20:09:25.298+0000] {main_pipeline.py:173} INFO - Writing batch with data: None to PostgreSQL.
[2025-02-02T20:09:25.673+0000] {main_pipeline.py:175} INFO - Batch 0 successfully written to PostgreSQL.
[2025-02-02T20:10:22.599+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T20:10:22.610+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T20:10:22.611+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T200916, end_date=20250202T201022
[2025-02-02T20:10:22.631+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T20:10:22.643+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T20:10:22.645+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-02-02T20:17:22.841+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-02T20:17:22.852+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T20:17:22.857+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [queued]>
[2025-02-02T20:17:22.858+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 6
[2025-02-02T20:17:22.866+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): spark_consumer> on 2025-01-28 00:00:00+00:00
[2025-02-02T20:17:22.870+0000] {standard_task_runner.py:72} INFO - Started process 579 to run task
[2025-02-02T20:17:22.872+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'save_to_minio', 'spark_consumer', 'scheduled__2025-01-28T00:00:00+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/main_pipeline.py', '--cfg-path', '/tmp/tmpewr71fo8']
[2025-02-02T20:17:22.874+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask spark_consumer
[2025-02-02T20:17:22.905+0000] {task_command.py:467} INFO - Running <TaskInstance: save_to_minio.spark_consumer scheduled__2025-01-28T00:00:00+00:00 [running]> on host b0b992709001
[2025-02-02T20:17:22.964+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='andrewsis' AIRFLOW_CTX_DAG_ID='save_to_minio' AIRFLOW_CTX_TASK_ID='spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-01-28T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-28T00:00:00+00:00'
[2025-02-02T20:17:22.966+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-02T20:17:28.168+0000] {main_pipeline.py:145} INFO - df_parsed является потоком
[2025-02-02T20:17:28.182+0000] {java_gateway.py:2273} INFO - Callback Server Starting
[2025-02-02T20:17:28.186+0000] {java_gateway.py:2275} INFO - Socket listening on ('127.0.0.1', 33731)
[2025-02-02T20:17:30.154+0000] {clientserver.py:561} INFO - Python Server ready to receive messages
[2025-02-02T20:17:30.156+0000] {clientserver.py:575} INFO - Received command c on object id p0
[2025-02-02T20:17:30.239+0000] {main_pipeline.py:169} INFO - Batch 0 пустой.
[2025-02-02T20:17:30.259+0000] {main_pipeline.py:174} INFO - Writing batch 0 to PostgreSQL.
[2025-02-02T20:17:30.515+0000] {main_pipeline.py:175} INFO - Writing batch with data: None to PostgreSQL.
[2025-02-02T20:17:30.841+0000] {main_pipeline.py:177} INFO - Batch 0 successfully written to PostgreSQL.
[2025-02-02T20:18:28.426+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-02T20:18:28.435+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-02T20:18:28.437+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=save_to_minio, task_id=spark_consumer, run_id=scheduled__2025-01-28T00:00:00+00:00, execution_date=20250128T000000, start_date=20250202T201722, end_date=20250202T201828
[2025-02-02T20:18:28.460+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-02T20:18:28.471+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-02T20:18:28.473+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
